unified-scheduler-pool/src/lib.rs

```diff
@@ -437,9 +444,38 @@ impl TaskHandler for DefaultTaskHandler {
         let index = task.task_index();
 
         let batch = bank.prepare_unlocked_batch_from_single_tx(transaction);
+        let transaction_indexes = match scheduling_context.mode() {
+            BlockVerification => vec![index],
+            BlockProduction => {
+                let mut vec = vec![];
+                if handler_context.transaction_status_sender.is_some() {
```

Comment:
I don't understand this part, could you explain what the thought process here is?

What does block-production have to do with `transaction_status_sender`; in nearly all cases it **should** be `None` for block-producers since it is only used for RPCs.

Reply:
Given below code, I'm thinking it may have been a typo and expected to be `transaction_recorder`?
Even so, it seems better to guarantee the recorder is present if we have block-production mode selected?

Reply:
Okay; I see now it's RPC-only stuff for the status batches.

Reply:
thanks for pointing out unclear code...: a59e39a6a8336f5fddf51171da848262f7ff6a78